### DISCUSSION TITLE
Extension of kibana_discover_app_url

### DISCUSSION THREAD

**Discussion by bonkotu** [jertel/elastalert2#583] [Q&A] [answered]
Hello,

I would like to add the time frame and filter set in Rule to the Kibana URL query generated by Elastalert.
Do you have any ideas?

**Comment by jertel** [comment, marked as answer]
See the Kibana link configuration in the ElastAlert 2 [documentation](https://elastalert2.readthedocs.io/en/latest/ruletypes.html?highlight=kibana#generate-kibana-link).

**Reply by bonkotu** [reply]
@jertel

Thank you for your reply. Sorry, I missed it when I saw the documentation.

Thanks to that I found the time settings in the docs.
However, for filtering, I haven't found a setting that works for me. I understood that the current Elastalert does not allow adding query keys to URLs in Kibana 7.x version. Is my understanding correct?

**Reply by jertel** [reply]
It looks like the Kibana discover URL supports query key filters in some capacity, according to [kibana_discover.py](https://github.com/jertel/elastalert2/blob/2bdd36a8c638667fa1f10130e72be178ec2b803f/elastalert/kibana_discover.py#L138).

However, the Kibana dashboard URL appears to only support query_string, term, and range filters, according to [kibana.py](https://github.com/jertel/elastalert2/blob/2bdd36a8c638667fa1f10130e72be178ec2b803f/elastalert/kibana.py#L212).

Personally, I don't use either in my alerts so I'm not able to offer much assistance on this.
